### ISSUE TITLE
Email File with HtmlBodyFilePath

### ISSUE THREAD

**Issue by radovandokic** [Microsoft/AL#2985]
Hi all,

Since we are working with streams in BC (with extensions V2 we cannot save files), what would be workaround for sending email with body layout?

Native function used in this case is in Codeunit 260 (Document-Mailing):
EmailFile(AttachmentFilePath : Text[250];AttachmentFileName : Text[250];**HtmlBodyFilePath : Text[250]**;PostedDocNo : Code[20];ToEmailAddress : Text[250];EmailDocName : Text[250];HideDialog : Boolean;ReportUsage : Integer) : Boolean

There is also function EmailFileFromStream(AttachementStream : InStream;AttachmentName : Text;Body : Text;Subject : Text;MailTo : Text;HideDialog : Boolean;ReportUsage : Integer) : Boolean
but it accepts body parameter as a text.

Basically, we have Custom Report Layout which we want to use as an email body.

Feedback is welcome.

Thanks,
Radovan

**Comment by radovandokic** [comment]
I forgot to mention, email has attachment.

**Comment by StanislawStempin** [comment]
Can you read the custom report layout into a stream, convert it to text and pass into the EmailFileFromStream function or is it too big? 
We might need to change the Body parameter type to an InStream instead of Text.

**Comment by radovandokic** [comment]
Not sure how to read the custom report layout into a stream. I can not use REPORT.SAVEAS since first parameter request integer and Custom report layout PK is Code 20.

**Comment by radovandokic** [comment]
Hi @StanislawStempin, would you please explain/help how to read the custom report layout into a stream?

**Comment by radovandokic** [comment]
Hi @StanislawStempin I tried to solve this issue with code like this:
....
CustomReportLayout.CalcFields(Layout);
CustomReportLayout.Layout.CreateInStream(InStr2);
while not InStr2.EOS() do 
  InStr2.ReadText(BodyText);  

and I use BodyText as a parameter for function DocumentMailing.EmailFileFromStream(...)
It seems that body is to big to fit into text variable.
Can you change the Body parameter type to an InStream instead of Text in DocumentMailing.EmailFileFromStream?

Thanks,
Radovan

**Comment by StanislawStempin** [comment]
What error are you getting when trying to invoke DocumentMailing.EmailFrilFromStream with the BodyText?

**Comment by radovandokic** [comment]
Hi,

I changed code a bit, and I am able to read stream into text, and to pass text to DocumentMailing.EmailFrilFromStream, but email body is not readable:

![image](https://user-images.githubusercontent.com/34686682/43644068-aecb6256-972d-11e8-8e43-c80d711852bd.png)

**Comment by StanislawStempin** [comment]
It looks like you are passing binary content into the body. Can you share the AL code that you're using for creating the stream and for calling EmailFileFromStream

**Comment by StanislawStempin** [comment]
Ok, so it does look like you're directly placing a .docx file inside the email body. The result seems expected in this case.
What is the behavior that you would like to achieve? Was there a file-based functionality before that rendered .docx report layouts correclty in an email body?

**Comment by radovandokic** [comment]
We would like to allow users to have Word document that will be used for email body. We have attachment in blob field, and we create stream and pass that attachment to DocumentMailing.EmailFileFromStream. Next step is to create email body from Word layout. We didn't have this before, but we would like to use this feature.
